common/src/main/java/com/redhat/cloud/notifications/models/validation/HttpsNonInternalUrlValidator.java

```diff
@@ -0,0 +1,49 @@
+package com.redhat.cloud.notifications.models.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+
+public class HttpsNonInternalUrlValidator implements ConstraintValidator<HttpsNonInternalUrl, String> {
+
+    private static final String SCHEME_HTTPS = "https";
+
+    @Override
+    public void initialize(HttpsNonInternalUrl constraintAnnotation) {
+        ConstraintValidator.super.initialize(constraintAnnotation);
+    }
+
+    /**
+     * Checks that the URL is well-formed, that the URI syntax is correct, and finally that the host doesn't point to
+     * an internal IP.
+     * @param rawUrl the URL to check.
+     * @param constraintValidatorContext the validation context.
+     * @return true if the provided URL is a well-formed, HTTPS URL which points to an external IP.
+     */
+    @Override
+    public boolean isValid(String rawUrl, ConstraintValidatorContext constraintValidatorContext) {
+        try {
+            final URL url = new URL(rawUrl);
+            final URI uri = url.toURI();
+
+            // Check that the scheme is "HTTPS".
+            if (!SCHEME_HTTPS.equals(uri.getScheme())) {
+                return false;
```

Comment:
We actually support non https calls - which probably we should warn the user in the UI instead of full blocking it.

Reply:
Great, no problem. Way better!